pymisp/api.py

```diff
@@ -389,6 +389,98 @@ def contact_event_reporter(self, event: Union[MISPEvent, int, str, UUID], messag
+        """
+        r = self._prepare_request('GET', f'eventReports/index/event_id:{event_id}')
+        event_reports = self._check_json_response(r)
+        if not (self.global_pythonify or pythonify) or 'errors' in event_reports:
+            return event_reports
+        to_return = []
+        for event_report in event_reports:
+            er = MISPEventReport()
+            er.from_dict(**event_report)
+            to_return.append(er)
+        return to_return
+
+    def add_event_report(self, event: Union[MISPEvent, int, str, UUID], event_report: MISPEventReport, pythonify: bool = False) -> Union[Dict, MISPEventReport]:
+        """Add an event report to an existing MISP event
+
+        :param event: event to extend
+        :param event_report: event report to add.
+        :param pythonify: Returns a PyMISP Object instead of the plain json output
+        """
+        event_id = get_uuid_or_id_from_abstract_misp(event)
+        r = self._prepare_request('POST', f'eventReports/add/{event_id}', data=event_report)
+        new_event_report = self._check_json_response(r)
+        if not (self.global_pythonify or pythonify) or 'errors' in new_event_report:
+            return new_event_report
+        er = MISPEventReport()
+        er.from_dict(**new_event_report)
+        return er
+
+    def update_event_report(self, event_report: MISPEventReport, event_report_id: Optional[int] = None, pythonify: bool = False) -> Union[Dict, MISPEventReport]:
+        """Update an event report on a MISP instance
+
+        :param event_report: event report to update
+        :param event_report_id: event report ID to update
+        :param pythonify: Returns a PyMISP Object instead of the plain json output
+        """
+        if event_report_id is None:
+            erid = get_uuid_or_id_from_abstract_misp(event_report)
+        else:
+            erid = get_uuid_or_id_from_abstract_misp(event_report_id)
+        r = self._prepare_request('POST', f'eventReports/edit/{erid}', data=event_report)
+        updated_event_report = self._check_json_response(r)
+        if not (self.global_pythonify or pythonify) or 'errors' in updated_event_report:
+            return updated_event_report
+        er = MISPEventReport()
+        er.from_dict(**updated_event_report)
+        return er
+
+    def delete_event_report(self, event_report: Union[MISPEventReport, int, str, UUID], hard: bool = False, pythonify: bool = False) -> Dict:
+        """Delete an event report from a MISP instance
+
+        :param event_report: event report to delete
+        :param hard: flag for hard delete
+        """
+        event_report_id = get_uuid_or_id_from_abstract_misp(event_report)
+        request_url = f'eventReports/delete/{event_report_id}'
+        if hard:
+            request_url += "/1"
+        r = self._prepare_request('POST', request_url)
+        response = self._check_json_response(r)
```

Comment:
I haven't tested yet, but does MISP returns a MISPEventReport on delete?

Reply:
MISP does return a MISPEventReport on soft delete. For a hard delete it does not (see line 475 for the addition of the `hard` param check

Reply:
alright, that's weird, and inconsistent with the other delete calls. @mokaddem is it expected?

Reply:
Any update on this one @mokaddem  CC @Rafiot

Reply:
@Rafiot I'm confused about the inconsistency you are mentioning.

Currently, **soft** deleting an Event report returns the Event report blob:
```json
{
    "EventReport": {
        "id": "x",
        "deleted": true
        ...
    }
}
```

**Hard** deleting a report returns the generic REST response:
```json
{
    "saved": true,
    "success": true,
    "name": "Event Report X hard deleted",
    "message": "Event Report X hard deleted",
    "url": "\/event_reports\/delete\/X"
}
```
What else would you expect? It feels weird to return the event report data as it has been deleted on the server.
For example, **hard** deleting an attribute returns the following:
```json
{
    "message": "Attribute deleted."
}
```

Reply:
Soft deleting something on MISP doesn't return the thing (at least it doesn't for attributes and events). It's not really a problem if deleting an event report does so, it's just different from the other calls.

Reply:
Ah the inconsistency was the other way around. Didn't get that.
It's changed https://github.com/MISP/MISP/commit/5e6444a663085c2f45e33fed2240f5e17e484b36

Reply:
Thank you!